src/pyHepGrid/src/test_nnlojob.py

```diff
@@ -44,5 +45,6 @@ def run_test(args, runcard):
 
     for r in rncards:
         nnlojob_args = testWrapper(r, dCards).replace("\"","").split()
-        util.spCall(["chmod","+x",header.runfile])
-        util.spCall(["./{0}".format(header.runfile)] + nnlojob_args)
+        runfile = os.path.basename(header.runfile)
+        util.spCall(["chmod","+x",runfile])
+        util.spCall(["./{0}".format(runfile)] + nnlojob_args)
```

Comment:
Question: what happens if `header.runfile` is not in the local directory?  (or is it always?)

Reply:
This is exactly what should changed. Before to run `test` mode you had to have the `runfile` in your current directory. Now you can also specify an absolute path. In the lines you highlighted we only get the file name the copy happens in: https://github.com/scarlehoff/pyHepGrid/blob/9fe0cf26da3c9bef827d98051adb296322920471/src/pyHepGrid/src/test_nnlojob.py#L15-L16

For the actual submission this already worked before, since we just pass `header.runfile` to Arc/Dirac/Slurm and they can handle absolute and relative paths.

Reply:
So does the `os.chdir` in `setup()` persist after it exits?

Reply:
What do you mean with "persist"? Within python yes, but it doesn't change your "main shell" from which you are running pyHepGrid. Nothing change in this regard. We also don't have to change back in `pyHepGrid` after `run_test`, since `pyHepGrid` finishes afterwards.

Reply:
Yes, within python - I assumed it would basically work like a local variable in the `setup` scope, but it looks like it does persist.